app/controllers/state_file/questions/questions_controller.rb

```diff
@@ -51,6 +52,15 @@ def redirect_if_no_intake
         end
       end
 
+      def redirect_if_intake_mismatch
+        if params[:current_intake].present?
+          if current_intake&.persisted? && current_intake.to_global_id.to_s != params[:current_intake]
```

Comment:
```suggestion
          if current_intake&.persisted? && current_intake.to_signed_global_id.to_s != params[:current_intake]
```
Can we sign these since they're exposed in the client ui?